src/points/PointsHistoryBottomSheet.tsx

```diff
@@ -0,0 +1,163 @@
+    dispatch(
+      getHistoryStarted({
+        getNextPage: false,
+      })
+    )
+  }
+
+  const onFetchMoreHistory = () => {
+    ValoraAnalytics.track(PointsEvents.points_screen_activity_fetch_more)
+    dispatch(
+      getHistoryStarted({
+        getNextPage: true,
+      })
+    )
+  }
+
+  const sections = useMemo(() => {
+    if (!pointsHistory.length || pointsHistoryStatus === 'error') {
+      return []
+    }
+    return groupFeedItemsInSections([], pointsHistory, (t: ClaimHistory) => Date.parse(t.createdAt))
+  }, [pointsHistory, pointsHistoryStatus])
+
+  // TODO: Render items
+  const renderItem = () => {
+    return <></>
+  }
+
+  const renderLoading = () => {
+    if (pointsHistoryStatus !== 'loading') {
+      return <></>
+    }
+    return (
+      <ActivityIndicator
+        testID={'PointsHistoryBottomSheet/Loading'}
+        style={styles.loadingIcon}
+        size="large"
+        color={colors.primary}
+      />
+    )
+  }
+
+  const renderEmpty = () => {
+    if (pointsHistoryStatus === 'error') {
+      return (
+        <View testID={'PointsHistoryBottomSheet/Error'} style={styles.errorContainer}>
+          <View style={styles.messageContainer}>
+            <Attention size={48} color={Colors.black} />
+            <Text style={styles.messageTitle}>{t('points.history.error.title')}</Text>
+            <Text style={styles.messageSubtitle}>{t('points.history.error.subtitle')}</Text>
+          </View>
+          <Button
+            testID={'PointsHistoryBottomSheet/TryAgain'}
+            onPress={onPressTryAgain}
+            text={t('points.history.error.tryAgain')}
+            type={BtnTypes.GRAY_WITH_BORDER}
+            size={BtnSizes.FULL}
+            style={{ width: '100%' }}
+          />
+        </View>
+      )
+    }
+
+    // TODO: Render empty/no history state
+    return <View testID={'PointsHistoryBottomSheet/Empty'}></View>
+  }
+
+  return (
+    <BottomSheetBase snapPoints={['80%']} forwardedRef={forwardedRef}>
+      <View style={styles.container}>
+        {pointsHistoryStatus !== 'error' && (
+          <Text style={styles.contentHeader}>{t('points.history.title')}</Text>
+        )}
+        <BottomSheetSectionList
+          contentContainerStyle={styles.contentContainer}
+          renderItem={renderItem}
+          renderSectionHeader={(item) => (
+            <SectionHead text={item.section.title} style={styles.sectionHead} />
+          )}
+          sections={sections}
+          keyExtractor={(item) => `${item.activity}-${item.createdAt}`}
+          keyboardShouldPersistTaps="always"
+          testID="PointsHistoryList"
+          onEndReached={onFetchMoreHistory}
+          ListFooterComponent={renderLoading}
+          ListEmptyComponent={renderEmpty}
+        />
+      </View>
+    </BottomSheetBase>
+  )
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    padding: Spacing.Thick24,
```

Comment:
Related to https://github.com/valora-inc/wallet/pull/5259#pullrequestreview-2008511618

There should be no padding here, otherwise the scroll indicator is placed incorrectly.
Also the scrollview needs to be fully scrollable without hitting an invisible wall at the bottom.
We should likely use a SafeAreaView or inset to properly handle this on devices with a home bar (iOS).

Reply:
Aha, after messing around with this, yeah -- your approach works well, with the appropriate padding applied to individual components (list entries and section headers), and using an inset for the bottom padding.